spec/models/user_spec.rb

```diff
@@ -1,9 +1,77 @@
 require 'rails_helper'
 
 describe User do
+	
+  before(:each) do
+    @user = create(:user)
+  end
 
   it { should belong_to :team }
   it { should belong_to :unit }
   it { should have_many :bills }
+  it { should have_many :invitations }
+  it { should have_many :senders }
+  it { should validate_length_of(:phone).is_equal_to(10), if: :phone }
+  it { should validate_numericality_of(:phone).only_integer }
+  
+  it 'has a valid factory' do
+    expect(@user).to be_valid
+  end
 
+  describe '.area_code' do
+    it 'should return area code' do
+      expect(@user.area_code).to eq '123'
+    end
+  end
+
+  describe '.exchange' do
+    it 'should return exchange' do
+      expect(@user.exchange).to eq '456'
+    end
+  end
+
+  describe '.line' do
+    it 'should return line' do
+     expect(@user.line).to eq '7890'
+    end
+  end
+
+  describe '.phone_string' do
+    it 'should return phone number with dashes inserted' do
+      expect(@user.phone_string).to eq '123-456-7890'
+    end
+  end
+
+  describe '.first_name_or_email' do
+ 		
+    context 'first name is nil' do
```

Comment:
This is also sort of like testing a private method directly. I think a better way to test this would be to actually take the actions of  `#email_without_domain` in the test suite itself:

`email_without_domain = @user.email.split('@').first`
`expect(@user.first_name_or_email).to eq email_without_domain`